src/Microsoft.PowerShell.Commands.Management/commands/management/Computer.cs

```diff
@@ -669,213 +355,347 @@ private void ProcessWSManProtocolForTestConnection()
+        /// Specify the Wait parameter. Prompt will be blocked is the Timeout is not 0
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        public SwitchParameter Wait { get; set; }
+
+        /// <summary>
+        /// Specify the Timeout parameter.
+        /// Negative value indicates wait infinitely.
+        /// Positive value indicates the seconds to wait before timeout.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        [Alias("TimeoutSec")]
+        [ValidateRange(-1, int.MaxValue)]
+        public int Timeout
+        {
+            get { return _timeout; }
+            set
             {
-                ArgumentException Ex = new ArgumentException(StringUtil.Format(ComputerResources.NoSystemDrive));
-                WriteError(new ErrorRecord(Ex, "EnableComputerNoSystemDrive", ErrorCategory.InvalidArgument, null));
+                _timeout = value;
+                _timeoutSpecified = true;
             }
-        }//end of BeginProcessing
+        }
+        private int _timeout = -1;
+        private bool _timeoutSpecified = false;
 
         /// <summary>
-        /// to implement ^C
+        /// Specify the For parameter.
+        /// Wait for the specific service before unblocking the prompt.
         /// </summary>
-        protected override void StopProcessing()
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        public WaitForServiceTypes For
         {
-            if (WMIClass != null)
+            get { return _waitFor; }
+            set
             {
-                WMIClass.Dispose();
+                _waitFor = value;
+                _waitForSpecified = true;
             }
         }
+        private WaitForServiceTypes _waitFor = WaitForServiceTypes.PowerShell;
+        private bool _waitForSpecified = false;
 
-#endregion Overrides
-    }//end of class
-#endregion
+        /// <summary>
+        /// Specify the Delay parameter.
+        /// The specific time interval (in second) to wait between network pings or service queries.
+        /// </summary>
+        [Parameter(ParameterSetName = DefaultParameterSet)]
+        [ValidateRange(1, Int16.MaxValue)]
+        public Int16 Delay
+        {
+            get { return (Int16)_delay; }
+            set
+            {
+                _delay = value;
+                _delaySpecified = true;
+            }
+        }
+        private int _delay = 5;
+        private bool _delaySpecified = false;
 
-#region Disable-ComputerRestore
+        /// <summary>
+        /// Script to test if the PowerShell is ready
+        /// </summary>
+        private const string TestPowershellScript = @"
+$array = @($input)
+$result = @{}
+foreach ($computerName in $array[1])
+{
+    $ret = $null
+    if ($null -eq $array[0])
+    {
+        $ret = Invoke-Command -ComputerName $computerName {$true} -SessionOption (New-PSSessionOption -NoMachineProfile) -ErrorAction SilentlyContinue
```

Comment:
Should we update this script to use splatting so we don't have to repeat most of the parameters?

Reply:
fixed